test/controllers/drivers_controller_test.rb

```diff
@@ -108,38 +142,74 @@
       # Ensure there is an existing driver saved
       # Assign the existing driver's id to a local variable
       # Set up the form data
+      driver = Driver.create(name: "Adell Jacobs", vin: "RF5J464C70D9C3KTB")
+      driver_id = driver.id
+
+      update_hash = {
+        driver: {
+          name: "Adell Jacobs",
+          vin: "RF5J464C70D9C3KTB",
+        },
+      }
 
       # Act-Assert
       # Ensure that there is no change in Driver.count
+      expect {
+        patch driver_path(driver_id), params: update_hash
+      }.wont_change "Driver.count"
 
       # Assert
       # Use the local variable of an existing driver's id to find the driver again, and check that its attributes are updated
+      expect(driver.name).must_equal update_hash[:driver][:name]
+      expect(driver.vin).must_equal update_hash[:driver][:vin]
       # Check that the controller redirected the user
-
+      must_respond_with :redirect
     end
 
     it "does not update any driver if given an invalid id, and responds with a 404" do
       # Arrange
       # Ensure there is an invalid id that points to no driver
       # Set up the form data
+      invalid_id = -1
+
+      update_hash = {
+        driver: {
+          name: "Adell Jacobs",
+          vin: "RF5J464C70D9C3KTB",
+        },
+      }
 
       # Act-Assert
       # Ensure that there is no change in Driver.count
+      expect {
+        patch driver_path(invalid_id), params: update_hash
+      }.wont_change "Driver.count"
 
       # Assert
       # Check that the controller gave back a 404
-
+      # must_respond_with :not_found
     end
 
     it "does not create a driver if the form data violates Driver validations, and responds with a redirect" do
       # Note: This will not pass until ActiveRecord Validations lesson
       # Arrange
+      driver = Driver.create(name: "Adell Jacobs", vin: "RF5J464C70D9C3KTB")
       # Ensure there is an existing driver saved
       # Assign the existing driver's id to a local variable
+      driver_id = driver.id
       # Set up the form data so that it violates Driver validations
+      update_hash = {
+        driver: {
+          name: nil,
+          vin: nil,
+        },
+      }
 
       # Act-Assert
       # Ensure that there is no change in Driver.count
+      expect {
+        patch driver_path(driver_id), params: update_hash
+      }.wont_change "Driver.count"
 
       # Assert
       # Check that the controller redirects
```

Comment:
Missing redirect test, and test that the driver still has the same name and vin.